### PR TITLE
Blog post for Bazel 2.1

### PR DESCRIPTION
**Pull request by laurentlb**

### CODE REVIEW

File: _posts/2020-02-06-bazel-2.1.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
+---
+layout: posts
+title: "Bazel 1.1"
+authors:
+  - laurentlb
+---
+
+[Bazel 2.1](https://github.com/bazelbuild/bazel/releases/tag/2.1.0) has just been released.
+
+Bazel 2.1 is intended to be backward-compatible with Bazel 2.0. However it contains a single incompatible change that was deemed a bug fix and thus eligible for inclusion. We haven’t found any code broken by this change, but please report if it causes any burden to you.
```
Comment (jin) [inline _posts/2020-02-06-bazel-2.1.md]:
```suggestion
Bazel 2.1 is intended to be backward-compatible with Bazel 2.0. However it contains a single incompatible change that was deemed a bug fix and thus eligible for inclusion. We haven’t found any code broken by this change, but [please report](https://github.com/bazelbuild/bazel/issues/new) if it causes any burden to you.
```

File: _posts/2020-02-06-bazel-2.1.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    same attribute value regardless of whether it is accessed as a field or
+    called like a method. Any arguments to the call are evaluated after the
+    `x.f` operation.
+*   Keyword-only arguments are now allowed:
+```python
+def foo(a, *, k): pass
+def foo(a, *b, k): pass
+foo(5, k = 6) # k can be specified only by keyword
+```
+*   Adds [`ctx.target_platform_has_constraint`](https://docs.bazel.build/versions/master/skylark/lib/ctx.html#target_platform_has_constraint) to allow rules to check the target platform's constraints.
```
Comment (jin) [inline _posts/2020-02-06-bazel-2.1.md]:
```suggestion
*   Added the [`ctx.target_platform_has_constraint`] boolean field (https://docs.bazel.build/versions/master/skylark/lib/ctx.html#target_platform_has_constraint) to allow rules to check the target platform's constraints.
```

File: _posts/2020-02-06-bazel-2.1.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+
+
+*   Similar to the
+    [.bazelignore](https://docs.bazel.build/versions/master/guide.html#.bazelignore)
+    in the main repository, a .bazelignore file in external repository will
+    cause the specified directories to be ignored by Bazel. Bazel won't try to
+    identify any packages under the directories, but the files can still be
+    referenced in other BUILD files.
+*   bazelignore files now support line comments, e.g. `# this is a comment`.
```
Comment (jin) [inline _posts/2020-02-06-bazel-2.1.md]:
```suggestion
*   .bazelignore files now support line comments, e.g. `# this is a comment`.
```

File: _posts/2020-02-06-bazel-2.1.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+
+## Android
+
+
+
+*   Experimental support for d8 merger is now available for use using
+    `--define=android_dexmerger_tool=d8_dexmerger`.
+*   Add desugaring support for `RoleManager#addRoleHolderAsUser`.
```
Comment (jin) [inline _posts/2020-02-06-bazel-2.1.md]:
```suggestion
*   Added desugaring support for `RoleManager#addRoleHolderAsUser`.
```

File: _posts/2020-02-06-bazel-2.1.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    identify any packages under the directories, but the files can still be
+    referenced in other BUILD files.
+*   bazelignore files now support line comments, e.g. `# this is a comment`.
+
+
+## Android
+
+
+
+*   Experimental support for d8 merger is now available for use using
```
Comment (jin) [inline _posts/2020-02-06-bazel-2.1.md]:
```suggestion
*   Added experimental support for merging dex files with D8 with
```

File: _posts/2020-02-06-bazel-2.1.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    `--define=android_dexmerger_tool=d8_dexmerger`.
+*   Add desugaring support for `RoleManager#addRoleHolderAsUser`.
+
+
+## Other changes
+
+
+
+*   Total critical path runtime will not be longer than total execution runtime.
```
Comment (jin) [inline _posts/2020-02-06-bazel-2.1.md]:
```suggestion
*   Fixed an issue with build performance profiles where the total critical path runtime will not be longer than total execution runtime.
```

File: _posts/2020-02-06-bazel-2.1.md [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+*   A new
```
Comment (jin) [inline _posts/2020-02-06-bazel-2.1.md]:
```suggestion
*   Added a new 
```

File: _posts/2020-02-06-bazel-2.1.md [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+    [`BuildEventProtocol`](https://docs.bazel.build/versions/master/build-event-protocol.html)
+    message is added for listing the convenience symlinks that get created in
```
Comment (jin) [inline _posts/2020-02-06-bazel-2.1.md]:
```suggestion
    message for listing the convenience symlinks created in
```

File: _posts/2020-02-06-bazel-2.1.md [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+    the workspace (such as `bazel-out`). It is enabled by
```
Comment (jin) [inline _posts/2020-02-06-bazel-2.1.md]:
```suggestion
    the workspace (such as `bazel-out`). Enable with the
```

File: _posts/2020-02-06-bazel-2.1.md [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+    `--experimental_convenience_symlinks_bep_event`.
```
Comment (jin) [inline _posts/2020-02-06-bazel-2.1.md]:
```suggestion
    `--experimental_convenience_symlinks_bep_event` flag.
```

File: _posts/2020-02-06-bazel-2.1.md [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+*   `cquery 'somepath` returns more reliable results when the dep has a
```
Comment (jin) [inline _posts/2020-02-06-bazel-2.1.md]:
```suggestion
*   `cquery 'somepath(...)'` returns more reliable results when the dependency has a
```

File: _posts/2020-02-06-bazel-2.1.md (hunk 4 above)
Comment (laurentlb) [reply _posts/2020-02-06-bazel-2.1.md]: Serge's recommendation was to use present tense (https://www.bazel.build/release-notes.html#guidelines).

**Comment by laurentlb** [comment]
PTAL

File: _posts/2020-02-06-bazel-2.1.md [hunk 12]
```diff
@@ -0,0 +1,87 @@
+---
+layout: posts
+title: "Bazel 1.1"
```
[the remark this answers is not in the document]
Comment (laurentlb) [reply _posts/2020-02-06-bazel-2.1.md]: Good catch, thanks :)

**Pull request merged**